app/controllers/carto/api/public/datasets_controller.rb

```diff
@@ -59,49 +51,56 @@ def enrich_tables(tables)
           table_names = tables.map { |table| table[:name] }
           visualizations = table_visualizations(table_names)
           tables.map do |table|
-            visualizations.find { |visualization| visualization[:name] == table[:name] } || table
+            table[:shared] = true unless table[:table_schema]==@user.username
+            viz = visualizations.find { |visualization| visualization[:name] == table[:name] } || { cartodbfied: false}
+            viz.merge(table)
           end
         end
```

Comment:
It seems you are not returning the `privacy` or `updated_at` for non cartodbfied tables. Also, I think it's better to avoid modifying the original parameter (`tables`). What about this other approach?
```suggestion
            viz = visualizations.find { |visualization| visualization[:name] == table[:name] }
            extra_fields = viz || default_extra_fields
            table.merge(extra_fields)
          end
        end

        def default_extra_fields
          {
            cartodbfied: false,
            shared: false,
            privacy: nil,
            updated_at: nil
          }
        end        
```